sdk/core/src/content_encoding.rs

```diff
@@ -0,0 +1,20 @@
+use crate::AddAsHeader;
+use http::request::Builder;
+
+#[derive(Debug, Clone, Copy)]
+pub struct ContentEncoding<'a>(&'a str);
+
+impl<'a> AddAsHeader for ContentEncoding<'a> {
+    fn add_as_header(&self, builder: Builder) -> Builder {
+        builder.header(http::header::CONTENT_ENCODING, self.0)
+    }
+}
+
+impl<'a, S> From<S> for ContentEncoding<'a>
+where
+    S: Into<&'a str>,
```

Comment:
I'm not sure how to make it work. 

If we modify the code to be like this:

```rust
  impl<'a, S> From<S> for ContentEncoding<'a>
  where
      S: AsRef<str> + 'a, 
  {
      fn from(s: S) -> Self {
        Self(s.as_ref())
      }   
  }
```

The compiler complains:

```bash
error[E0515]: cannot return value referencing function parameter `s`
  --> sdk/core/src/content_encoding.rs:18:9
   |
18 |         Self(s.as_ref())
   |         ^^^^^-^^^^^^^^^^
   |         |    |
   |         |    `s` is borrowed here
   |         returns a value referencing data owned by the current function
```

We could avoid to call `as_ref()` and keep the generic but it would complicate the type signature. I'm not sure it's worth the change. 

This will become:

```rust
#[derive(Debug, Clone, Copy)]
pub struct ContentEncoding<S: AsRef<str>>(S);

impl<S: AsRef<str>> AddAsHeader for ContentEncoding<S> { 
    fn add_as_header(&self, builder: Builder) -> Builder { 
        builder.header(http::header::CONTENT_ENCODING, self.0.as_ref())
    } 
}

impl<S: AsRef<str>> From<S> for ContentEncoding<S>
where
    S: AsRef<str>,
{
    fn from(s: S) -> Self { 
        Self(s)
    } 
}
```

Reply:
No you're right. Nevermind...